### PR TITLE
DataTypeDefinition CRUD operations included as part of CDL package

### PR DESCRIPTION
**Pull request by roopesh83**

### CODE REVIEW

File: cdl/dtd_service.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import (
+	"fmt"
+	"github.com/go-playground/validator/v10"
+	"io"
+	"net/http"
+	"path"
+)
+
+type DataTypeDefinition struct {
```
Comment (loafoe) [inline cdl/dtd_service.go:11]: Please run `go fmt` and commit

File: cdl/dtd_service.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	ID string `json:"id,omitempty"`
+	Name string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	JsonSchema interface{} `json:"jsonSchema,omitempty"`
```
Comment (loafoe) [inline cdl/dtd_service.go:15]: instead of `interface{}` you can use the `json.RawMessage` type here

**Review by loafoe — CHANGES_REQUESTED** [review]
great!! just 2 minor change requests and we should be good to merge

File: cdl/dtd_service.go (hunk 1 above)
Comment (roopesh83) [reply cdl/dtd_service.go:12]: "go fmt " command has been run, the code in dtd_service.go, dtd_service_test.go and cdl/client.go have been formatted.

File: cdl/dtd_service.go (hunk 2 above)
Comment (roopesh83) [reply cdl/dtd_service.go]: Now using json.RawMessage for jsonSchema field of DTD

**Pull request merged**